File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [[False] * n for _ in range(n)]
    q = deque()

    for i in range(n):
        for j in range(n):

            if computers[i][j] and not visited[i][j]:
                q.append((i,j))

                while q :
                    x, y = q.popleft()
                    visited[x][y] = True
                    for _ in range(n):
                        if computers[x][_] and not visited[x][_]:
                            visited[x][_] = True
                            q.append((_,x))
                answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for(int i=0; i < n; i++){
            if(!visited[i]){
                dfs(i, computers, visited);
                answer++;
            }    
        }
        
        
        return answer;
    }
    
    public void dfs(int current, int[][]computers, boolean[] visited){
        visited[current] = true;
        
        for(int next = 0; next < computers.length; next ++){
            if(current != next && computers[current][next] == 1 && !visited[next]){
                dfs(next, computers, visited);
            }
        }
    }
    
    
}